generic/image.c

```diff
@@ -1480,13 +1550,14 @@ int image_(Main_lab2rgb)(lua_State *L) {
       b =  0.0556434 * X - 0.2040259 * Y + 1.0572252 * Z;
 
       // set rgb
-      THTensor_(set3d)(rgb, 0, y, x, gamma_compress_sRGB(r));
-      THTensor_(set3d)(rgb, 1, y, x, gamma_compress_sRGB(g));
-      THTensor_(set3d)(rgb, 2, y, x, gamma_compress_sRGB(b));
+      THTensor_(set3d)(rgb, 0, y, x, image_(gamma_compress_sRGB(r)));
+      THTensor_(set3d)(rgb, 1, y, x, image_(gamma_compress_sRGB(g)));
+      THTensor_(set3d)(rgb, 2, y, x, image_(gamma_compress_sRGB(b)));
     }
   }
   return 0;
 }
+#endif /*TH_REAL_IS_BYTE*/
```

Comment:
To avoid errors like `attempt to call field 'rgb2lab' (a nil value)` if the caller makes the mistake to call these function on a byte tensor a specific error message might help? e.g.:

```Lua
int image_(Main_rgb2lab)(lua_State *L) {
  return luaL_error(L, "image.rgb2lab: not supported for torch.ByteTensor");
}

int image_(Main_lab2rgb)(lua_State *L) {
  return luaL_error(L, "image.lab2rgb: not supported for torch.ByteTensor");
}
```